In [1]:
import pandas as pd
import os

os.chdir("/Users/User/OneDrive/Documents/MSc Data Science/BI and visualisation/Data Set")
print(os.getcwd())

C:\Users\User\OneDrive\Documents\MSc Data Science\BI and visualisation\Data Set


since pandas did not recognise question marks and name error codes as null values, the na_values parameter was used below. These special cases were introduced as null values in addition to the null values already recognised by pandas. 

In [2]:
#list of missing value types to be recognised
missing_val = ['?','#NAME?']


#Reading and converting csv data in to data frame
df_retail_sales=pd.read_csv('Project 2 - Retail Sales Dataset.csv',skipinitialspace = True, na_values = missing_val)
df_retail_sales

ORDER_NUMBER  SHIPMENT_NUMBER  ITEM_NUMBER INVOICE_NUMBER  INVOICE_DATE  \
0         6735217         20171129       428714        6735217      20171129   
1         6165708         20190626       408872        6165708      20190626   
2         6517073         20180910       354274        6517073      20180910   
3         6723693         20200513       222679        6723693      20200513   
4         6766514         20190128       431999        6766514      20190128   
...           ...              ...          ...            ...           ...   
9995      5100802         20190403       522110        5100802      20190403   
9996      5462559         20200827      4745048        5462559      20200827   
9997      5165659         20190708       620658        5165659      20190708   
9998      3274935         20200801       177550        3274935      20200801   
9999      6104799         20201129       398867        6104799      20201129   

      ORDER_DATE  CUSTOMER_NUMBER  CUSTOMER_LOCATION  CUSTOMER_DEPARTMENT  \
0       20171128              410                 10                    9   
1       20190624             1375                  1                   83   
2       20180908             2196                  1                   70   
3       20200509             1934                 50                   70   
4       20190125            75379                201                    0   
...          ...              ...                ...                  ...   
9995    20190329             1560                  4                   30   
9996    20200825            75161                  1                    0   
9997    20190707             1954                  1                   62   
9998    20200731             2486                  4                   66   
9999    20201123            15165                  1                    0   

     CUSTOMER_PROVINCE  ... COMPANY_CODE       COMPANY_NAME COMPANY_NUMBER  \
0                TEXAS  ...        88880  BRANDS GLOBAL CO.              0   
1              ARIZONA  ...        88880  BRANDS GLOBAL CO.              0   
2           CALIFORNIA  ...        88880  BRANDS GLOBAL CO.              0   
3              ARIZONA  ...        88880  BRANDS GLOBAL CO.              0   
4                TEXAS  ...        88880  BRANDS GLOBAL CO.              0   
...                ...  ...          ...                ...            ...   
9995          MICHIGAN  ...        88880  BRANDS GLOBAL CO.            513   
9996          MICHIGAN  ...        88880  BRANDS GLOBAL CO.            534   
9997          MICHIGAN  ...        88880  BRANDS GLOBAL CO.            528   
9998             TEXAS  ...        88880  BRANDS GLOBAL CO.              0   
9999        CALIFORNIA  ...        72223  THE GROCERY GROUP              0   

     AVERAGE_COST_PRICE SALES_AMOUNT  INVOICE_AMOUNT  COST_PRICE  \
0                   0.0         3.36            0.56        0.56   
1                   0.0        12.00            4.00        4.00   
2                   0.0         7.16            1.79        1.79   
3                   0.0        19.60            4.90        4.90   
4                   0.0        62.58           10.43       10.43   
...                 ...          ...             ...         ...   
9995                0.0         0.00            0.00        0.00   
9996                0.0         0.00            0.00        0.00   
9997                0.0         0.00            0.00        0.00   
9998                0.0         0.00            0.00        0.00   
9999                0.0         0.00            0.00        0.00   

        UPC_NUMBER  PURCHASE_ORDER_NUMBER  ORDER_TYPE  
0     6.224543e+10                    0.0     REGULAR  
1     6.224543e+10                    0.0     REGULAR  
2     6.224543e+10                    0.0     REGULAR  
3     6.224543e+10                    0.0     REGULAR  
4     6.224543e+10                    0.0     REGULAR  
...            ...           

Column headings, total number of rows and columns, data type of each column, total memory usage and the count of non-null values of each column of this dataset was analysed using the info() method. 

In [3]:
#Summary of the dataframe
df_retail_sales.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10000 entries, 0 to 9999
Data columns (total 46 columns):
 #   Column                      Non-Null Count  Dtype  
---  ------                      --------------  -----  
 0   ORDER_NUMBER                10000 non-null  object 
 1   SHIPMENT_NUMBER             10000 non-null  int64  
 2   ITEM_NUMBER                 10000 non-null  int64  
 3   INVOICE_NUMBER              10000 non-null  object 
 4   INVOICE_DATE                10000 non-null  int64  
 5   ORDER_DATE                  10000 non-null  int64  
 6   CUSTOMER_NUMBER             10000 non-null  int64  
 7   CUSTOMER_LOCATION           10000 non-null  int64  
 8   CUSTOMER_DEPARTMENT         10000 non-null  int64  
 9   CUSTOMER_PROVINCE           9948 non-null   object 
 10  ITEM_DESCRIPTION            9928 non-null   object 
 11  SHIPPING_REGION_IDENTIFIER  10000 non-null  int64  
 12  SHIPPING_REGION_CODE        10000 non-null  object 
 13  SHIPPING_REGION_NAME        1000

The results of info() method showed that there are null values in some of the columns in the dataset, so the isnull().sum() method was used to get a count of the null values of each column. 

In [4]:
#Checking for any null Values
df_retail_sales.isnull().sum()

ORDER_NUMBER                     0
SHIPMENT_NUMBER                  0
ITEM_NUMBER                      0
INVOICE_NUMBER                   0
INVOICE_DATE                     0
ORDER_DATE                       0
CUSTOMER_NUMBER                  0
CUSTOMER_LOCATION                0
CUSTOMER_DEPARTMENT              0
CUSTOMER_PROVINCE               52
ITEM_DESCRIPTION                72
SHIPPING_REGION_IDENTIFIER       0
SHIPPING_REGION_CODE             0
SHIPPING_REGION_NAME             0
SHIPPING_UNIT_MEASURE            0
SHIPPING_PACK_QUANTITY           0
ITEM_UNIT_QUANTITY               0
ITEM_DEPARTMENT                  0
ITEM_SECTION                     0
ITEM_SECTION_MODULE              0
ITEM_GROUP                       0
ITEM_SEQUENCE                    0
ITEM_STOCK_STATUS               52
WAREHOUSE_IDENTIFIER             0
WAREHOUSE_NUMBER                 0
WAREHOUSE_NAME                   0
WAREHOUSE_REGION_NAME            0
VENDOR_IDENTIFIER                0
VENDOR_NUMBER       

There were 574 missing values for VENDOR_NAME and VENDOR_NUMBER columns. However, VENDOR_IDENTIFIER had no null values so it was decided that for further study VENDOR_IDENTIFIER will be used as it served the same purpose of vendor identification.  

VENDOR_CITY, VENDOR_COUNTRY, VENDOR_ADDRESS columns were decided to be dropped as these had 2,095 null values for each. Also, it was decided that a detailed vendor analysis will not be done in this study.

52 rows had missing (blank) ITEM_STOCK_STATUS as well as for CUSTOMER_PROVINCE so these rows were dropped as it was only 0.52% of the total number of rows. Then the effect of this on the data set was checked using the .shape property of pandas library. The total number of rows were then reduced to 9948.

In [5]:
#Removing null values in certain columns
df_new = df_retail_sales.dropna(subset = ["CUSTOMER_PROVINCE","ITEM_STOCK_STATUS"])
df_new

#Shape of the new dataframe after removing null values
df_new.shape

(9948, 46)

ITEM_DESCIPTION had 72 null values but when checking for values in that column, it was found that the descriptions were complicated with numbers, letters and even symbols. It also had so many unique values.  So, it was decided that this column may not be used in visualisations. 

In [6]:
df_retail_sales.ITEM_DESCRIPTION.unique()

array(['600890 1-1/2 IN 45 DEG ELB(027241) ',
       'ABS COUPLER REDUCER 4 X 3-IN       ',
       'ABS TRAP ADAPTER MPT 1-1/2X1-1/4   ', ...,
       'SWISS EMMENTHAL- 3.3 KG            ',
       'FOIL CAKE PAN & LID SQUARE 8-IN 3PK',
       'CARE+ P/LINER EXT COVERGE CONTOUR  '], dtype=object)

In [7]:
df_retail_sales.ITEM_DESCRIPTION.nunique()

6706

Instead, it was decided that ITEM_DEPARTMENT will be used in visualisations regarding products, as it had no null values and it had categorical data with all products classified as 5 department numbers.

In [8]:
df_retail_sales.ITEM_DEPARTMENT.nunique()

5

In [9]:
df_retail_sales.ITEM_DEPARTMENT.unique()

array([40, 30, 20,  0, 10], dtype=int64)

Few other columns relating to product identification such as ITEM_SECTION, ITEM_NUMBER these were not planned to be used in further study since there were many unique values in each of those columns as shown below. 

In [10]:
df_retail_sales.ITEM_NUMBER.nunique()

6704

In [11]:
df_retail_sales.ITEM_SECTION.nunique()

76

In [12]:
df_retail_sales.ITEM_SECTION_MODULE.nunique()

72

In [13]:
df_retail_sales.ITEM_GROUP.nunique() 

760

In [14]:
df_retail_sales.ITEM_SEQUENCE.nunique() 

627

In [15]:
df_retail_sales.VENDOR_IDENTIFIER.nunique() 

1290

In [16]:
df_retail_sales.SHIPPING_REGION_NAME.unique() 

array(['FLORIDA', 'OHIO', 'TEXAS', 'MISSOURI'], dtype=object)

In [17]:
df_retail_sales.COMPANY_NAME.unique() 

array(['BRANDS GLOBAL CO.', 'THE GROCERY GROUP', 'THE PRODUCE CLUB'],
      dtype=object)

PURCHASE_ORDER_NUMBER had 7 missing values, but it was an irrelevant column for visualization purposes so that whole column was dropped. INVOICE_NUMBER were irrelevant for visualisation as well, so this column was also dropped. 

There were different ways of identification for different categories such as shipping region, warehouses, company, and vendor. For shipping region identification, we thought that the SHIPPING_REGION_NAME is ideal as it had four unique values only and had no null values. 

So, SHIPPING_REGION_IDENTIFIER, SHIPPING_REGION_CODE columns were decided to be dropped. Similarly, COMPANY_IDENTIFIER, COMPANY_CODE, COMPANY_NUMBER were dopped as COMPANY_NAME served the best for company identification with three unique values and no null values. 

In [18]:
#Removing certain columns that are not relevant
df_new.drop(["INVOICE_NUMBER","ITEM_DESCRIPTION","SHIPPING_REGION_IDENTIFIER","SHIPPING_REGION_CODE","WAREHOUSE_IDENTIFIER",
             "WAREHOUSE_NUMBER","COMPANY_IDENTIFIER","COMPANY_CODE","COMPANY_NUMBER","VENDOR_NUMBER","VENDOR_ADDRESS",
             "PURCHASE_ORDER_NUMBER"],axis=1,inplace=True)

C:\Users\User\AppData\Local\Temp\ipykernel_5876\1356719526.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_new.drop(["INVOICE_NUMBER","ITEM_DESCRIPTION","SHIPPING_REGION_IDENTIFIER","SHIPPING_REGION_CODE","WAREHOUSE_IDENTIFIER",


In [19]:
df_new.shape

(9948, 34)

In [20]:
#to see first few rows 
df_new.head()

ORDER_NUMBER  SHIPMENT_NUMBER  ITEM_NUMBER  INVOICE_DATE  ORDER_DATE  \
0      6735217         20171129       428714      20171129    20171128   
1      6165708         20190626       408872      20190626    20190624   
2      6517073         20180910       354274      20180910    20180908   
3      6723693         20200513       222679      20200513    20200509   
4      6766514         20190128       431999      20190128    20190125   

   CUSTOMER_NUMBER  CUSTOMER_LOCATION  CUSTOMER_DEPARTMENT CUSTOMER_PROVINCE  \
0              410                 10                    9             TEXAS   
1             1375                  1                   83           ARIZONA   
2             2196                  1                   70        CALIFORNIA   
3             1934                 50                   70           ARIZONA   
4            75379                201                    0             TEXAS   

  SHIPPING_REGION_NAME  ... VENDOR_COUNTRY  ORDER_QUANTITY  SHIPPED_QUANTITY  \
0              FLORIDA  ...             US               6                 6   
1              FLORIDA  ...             US               3                 3   
2              FLORIDA  ...             US               4                 4   
3              FLORIDA  ...             US               4                 4   
4              FLORIDA  ...             US               6                 6   

        COMPANY_NAME  AVERAGE_COST_PRICE  SALES_AMOUNT  INVOICE_AMOUNT  \
0  BRANDS GLOBAL CO.                 0.0          3.36            0.56   
1  BRANDS GLOBAL CO.                 0.0         12.00            4.00   
2  BRANDS GLOBAL CO.                 0.0          7.16            1.79   
3  BRANDS GLOBAL CO.                 0.0         19.60            4.90   
4  BRANDS GLOBAL CO.                 0.0         62.58           10.43   

   COST_PRICE    UPC_NUMBER ORDER_TYPE  
0        0.56  6.224543e+10    REGULAR  
1        4.00  6.224543e+10    REGULAR  
2        1.79  6.224543e+10    REGULAR  
3        4.90  6.224543e+10    REGULAR  
4       10.43  6.224543e+10    REGULAR  

[5 rows x 34 columns]

## Checking for any inconsistencies in Ware house name and Ware house Region Name

In [21]:
warehouse_list = list(zip(df_new["WAREHOUSE_NAME"].values, df_new["WAREHOUSE_REGION_NAME"].values))

inconsistent = []
warehouse_dict = {}

for name, region in warehouse_list:
    
  if name not in warehouse_dict:
    warehouse_dict[name] = region
  else:
    if warehouse_dict[name] != region:
      inconsistent.append(name)
    
#Checking if there are any inconsistent records in the inconsistent list
inconsistent

[]

There were no inconsistencies. 

## Change date columns to date format

Some of the columns with date was not in the date format but just integer format. These were changed to date format below. 

In [22]:
#Converting date columns to date time format
df_retail_sales['INVOICE_DATE']= pd.to_datetime(df_retail_sales['INVOICE_DATE'],format='%Y%m%d')
df_retail_sales['ORDER_DATE']= pd.to_datetime(df_retail_sales['ORDER_DATE'],format='%Y%m%d')

In [23]:
#Checking if the date columns has changed to date time format correctly
df_retail_sales

ORDER_NUMBER  SHIPMENT_NUMBER  ITEM_NUMBER INVOICE_NUMBER INVOICE_DATE  \
0         6735217         20171129       428714        6735217   2017-11-29   
1         6165708         20190626       408872        6165708   2019-06-26   
2         6517073         20180910       354274        6517073   2018-09-10   
3         6723693         20200513       222679        6723693   2020-05-13   
4         6766514         20190128       431999        6766514   2019-01-28   
...           ...              ...          ...            ...          ...   
9995      5100802         20190403       522110        5100802   2019-04-03   
9996      5462559         20200827      4745048        5462559   2020-08-27   
9997      5165659         20190708       620658        5165659   2019-07-08   
9998      3274935         20200801       177550        3274935   2020-08-01   
9999      6104799         20201129       398867        6104799   2020-11-29   

     ORDER_DATE  CUSTOMER_NUMBER  CUSTOMER_LOCATION  CUSTOMER_DEPARTMENT  \
0    2017-11-28              410                 10                    9   
1    2019-06-24             1375                  1                   83   
2    2018-09-08             2196                  1                   70   
3    2020-05-09             1934                 50                   70   
4    2019-01-25            75379                201                    0   
...         ...              ...                ...                  ...   
9995 2019-03-29             1560                  4                   30   
9996 2020-08-25            75161                  1                    0   
9997 2019-07-07             1954                  1                   62   
9998 2020-07-31             2486                  4                   66   
9999 2020-11-23            15165                  1                    0   

     CUSTOMER_PROVINCE  ... COMPANY_CODE       COMPANY_NAME COMPANY_NUMBER  \
0                TEXAS  ...        88880  BRANDS GLOBAL CO.              0   
1              ARIZONA  ...        88880  BRANDS GLOBAL CO.              0   
2           CALIFORNIA  ...        88880  BRANDS GLOBAL CO.              0   
3              ARIZONA  ...        88880  BRANDS GLOBAL CO.              0   
4                TEXAS  ...        88880  BRANDS GLOBAL CO.              0   
...                ...  ...          ...                ...            ...   
9995          MICHIGAN  ...        88880  BRANDS GLOBAL CO.            513   
9996          MICHIGAN  ...        88880  BRANDS GLOBAL CO.            534   
9997          MICHIGAN  ...        88880  BRANDS GLOBAL CO.            528   
9998             TEXAS  ...        88880  BRANDS GLOBAL CO.              0   
9999        CALIFORNIA  ...        72223  THE GROCERY GROUP              0   

     AVERAGE_COST_PRICE SALES_AMOUNT  INVOICE_AMOUNT  COST_PRICE  \
0                   0.0         3.36            0.56        0.56   
1                   0.0        12.00            4.00        4.00   
2                   0.0         7.16            1.79        1.79   
3                   0.0        19.60            4.90        4.90   
4                   0.0        62.58           10.43       10.43   
...                 ...          ...             ...         ...   
9995                0.0         0.00            0.00        0.00   
9996                0.0         0.00            0.00        0.00   
9997                0.0         0.00            0.00        0.00   
9998                0.0         0.00            0.00        0.00   
9999                0.0         0.00            0.00        0.00   

        UPC_NUMBER  PURCHASE_ORDER_NUMBER  ORDER_TYPE  
0     6.224543e+10                    0.0     REGULAR  
1     6.224543e+10                    0.0     REGULAR  
2     6.224543e+10                    0.0     REGULAR  
3     6.224543e+10                    0.0     REGULAR  
4     6.224543e+10                    0.0     REGULAR  
...            ...                    ...         ...

In [24]:
df_new.columns

Index(['ORDER_NUMBER', 'SHIPMENT_NUMBER', 'ITEM_NUMBER', 'INVOICE_DATE',
       'ORDER_DATE', 'CUSTOMER_NUMBER', 'CUSTOMER_LOCATION',
       'CUSTOMER_DEPARTMENT', 'CUSTOMER_PROVINCE', 'SHIPPING_REGION_NAME',
       'SHIPPING_UNIT_MEASURE', 'SHIPPING_PACK_QUANTITY', 'ITEM_UNIT_QUANTITY',
       'ITEM_DEPARTMENT', 'ITEM_SECTION', 'ITEM_SECTION_MODULE', 'ITEM_GROUP',
       'ITEM_SEQUENCE', 'ITEM_STOCK_STATUS', 'WAREHOUSE_NAME',
       'WAREHOUSE_REGION_NAME', 'VENDOR_IDENTIFIER', 'VENDOR_NAME',
       'VENDOR_CITY', 'VENDOR_COUNTRY', 'ORDER_QUANTITY', 'SHIPPED_QUANTITY',
       'COMPANY_NAME', 'AVERAGE_COST_PRICE', 'SALES_AMOUNT', 'INVOICE_AMOUNT',
       'COST_PRICE', 'UPC_NUMBER', 'ORDER_TYPE'],
      dtype='object')

## Checking for any inaccurate values for columns

The accuracy of the values of some columns were checked as shown below.That is, columns involving quantities and prices cannot be negative. So this was checked using the minimum and maximum values for these particular columns.

In [25]:
df_new["COST_PRICE"].min(), df_new["COST_PRICE"].max()

(0.0, 20148.4)

In [26]:
df_new["AVERAGE_COST_PRICE"].min(), df_new["AVERAGE_COST_PRICE"].max()

(0.0, 424.16)

In [27]:
df_new["SHIPPING_PACK_QUANTITY"].min(), df_new["SHIPPING_PACK_QUANTITY"].max()

(1, 1000)

In [28]:
df_new["ITEM_UNIT_QUANTITY"].min(), df_new["ITEM_UNIT_QUANTITY"].max()

(1.0, 2000.0)

In [29]:
df_new["ORDER_QUANTITY"].min(), df_new["ORDER_QUANTITY"].max()

(1, 1080)

In [30]:
df_new["SHIPPED_QUANTITY"].min(), df_new["SHIPPED_QUANTITY"].max()

(0, 1080)

In [31]:
df_new["SALES_AMOUNT"].min(), df_new["SALES_AMOUNT"].max()

(0.0, 25278.68)

## Checking the shape of the new dataset 

In [32]:
df_new.shape

(9948, 34)

In [33]:
#to see first few rows 
df_new.head()

ORDER_NUMBER  SHIPMENT_NUMBER  ITEM_NUMBER  INVOICE_DATE  ORDER_DATE  \
0      6735217         20171129       428714      20171129    20171128   
1      6165708         20190626       408872      20190626    20190624   
2      6517073         20180910       354274      20180910    20180908   
3      6723693         20200513       222679      20200513    20200509   
4      6766514         20190128       431999      20190128    20190125   

   CUSTOMER_NUMBER  CUSTOMER_LOCATION  CUSTOMER_DEPARTMENT CUSTOMER_PROVINCE  \
0              410                 10                    9             TEXAS   
1             1375                  1                   83           ARIZONA   
2             2196                  1                   70        CALIFORNIA   
3             1934                 50                   70           ARIZONA   
4            75379                201                    0             TEXAS   

  SHIPPING_REGION_NAME  ... VENDOR_COUNTRY  ORDER_QUANTITY  SHIPPED_QUANTITY  \
0              FLORIDA  ...             US               6                 6   
1              FLORIDA  ...             US               3                 3   
2              FLORIDA  ...             US               4                 4   
3              FLORIDA  ...             US               4                 4   
4              FLORIDA  ...             US               6                 6   

        COMPANY_NAME  AVERAGE_COST_PRICE  SALES_AMOUNT  INVOICE_AMOUNT  \
0  BRANDS GLOBAL CO.                 0.0          3.36            0.56   
1  BRANDS GLOBAL CO.                 0.0         12.00            4.00   
2  BRANDS GLOBAL CO.                 0.0          7.16            1.79   
3  BRANDS GLOBAL CO.                 0.0         19.60            4.90   
4  BRANDS GLOBAL CO.                 0.0         62.58           10.43   

   COST_PRICE    UPC_NUMBER ORDER_TYPE  
0        0.56  6.224543e+10    REGULAR  
1        4.00  6.224543e+10    REGULAR  
2        1.79  6.224543e+10    REGULAR  
3        4.90  6.224543e+10    REGULAR  
4       10.43  6.224543e+10    REGULAR  

[5 rows x 34 columns]

## Creating a column for profit, revenue and cost

New columns for revenue, cost and profit were made for better business insights for analysis later on. 

In [35]:
df_new["REVENUE"] = df_new["SALES_AMOUNT"] * df_new["ORDER_QUANTITY"]

C:\Users\User\AppData\Local\Temp\ipykernel_6224\759782189.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_new["REVENUE"] = df_new["SALES_AMOUNT"] * df_new["ORDER_QUANTITY"]


In [36]:
df_new["COST"] = df_new["COST_PRICE"] * df_new["ORDER_QUANTITY"]

C:\Users\User\AppData\Local\Temp\ipykernel_6224\3416835204.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_new["COST"] = df_new["COST_PRICE"] * df_new["ORDER_QUANTITY"]


In [37]:
df_new["PROFIT"] = df_new["REVENUE"] - df_new["COST"]

C:\Users\User\AppData\Local\Temp\ipykernel_6224\3960797594.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_new["PROFIT"] = df_new["REVENUE"] - df_new["COST"]


## Exporting the cleaned data to a csv file

In [34]:
#the new cleaned data set was exported to a new csv to use for future analysis and visualisations
df_new.to_csv('Clean Retail Sales Dataset.csv',index=False)